packages/pds-ember/addon/components/pds/checkbox-field/index.js

```diff
@@ -0,0 +1,47 @@
+import Component from '@glimmer/component'
+import { guidFor } from '@ember/object/internals'
+
+/**
+ * @class PdsCheckboxField
+ *
+ * @yield {Hash} field
+ * @yield {string} field.id auto-generated ID applied to rendered `input[id]`
+ * and `label[for]`
+ */
+
+/**
+ * Optional value to apply to `input[id]` and `label[for]` attributes in
+ * generated markup.  If not provided, an auto-generated guid will be applied.
+ *
+ * **NOTE:**
+ * This argument must be used if you want to apply a specific `input[id]` value.
+ *
+ * @argument id
```

Comment:
```suggestion
 * @argument [id]
```

Is it correct to assume these are intended to be JSDoc header comments? 🤔 Or are they a non-standard / unique-to-Ember DSL?

If they're JSDoc, I believe this line could represent optionality of the `id` argument using the suggested syntax. SEE: https://jsdoc.app/tags-param.html (`@argument` is an alias for `@param`)

If this comment block will be interpreted for IDE intellisense / auto-complete etc. then this could be a better developer experience downstream.

Reply:
`ember-cli-storybook` makes use of the comments to generate JSON metadata during the `ember build` phase, so that Storybook can load it at run time.  The comments are in [YUIDoc format](https://yui.github.io/yuidoc/syntax/index.html) (which is very similar to JSDoc syntax), but has some extra tags available (via [`ember-cli-addon-docs-yuidoc`](https://github.com/ember-learn/ember-cli-addon-docs-yuidoc/#usage)) to provide Storybook with useful metadata.